test_programs/execution_success/hashmap/src/main.nr

```diff
@@ -0,0 +1,150 @@
+use dep::std::collections::map::HashMap;
+use dep::std::hash::BuildHasherDefault;
+use dep::std::hash::pedersen::PedersenHasher;
+
+fn main() {
+    test_sequential();
+    test_multiple_equal_insert();
+    test_value_override();
+    test_insert_and_methods();
+    test_retain();
+    test_hashmaps_equality();
```

Comment:
Do you mind adding some inputs to the program and feeding those into the hash maps?
Without any program inputs the compiler can usually evaluate/simplify away the program at compile-time which could make some bugs harder to spot.